src/content/en/updates/2017/06/supercharged-codesplit.md

```diff
@@ -0,0 +1,281 @@
+splitting your code is to use route-based chunking.
+* **Route-based chunking**: Route-based chunking creates bundles that are related your app’s routes.
+By analyzing your routes and their dependencies, we can make decision what modules go into which
+bundle.
+
+### Why do code splitting?
+
+#### Loose modules
+
+With natives ES6 modules, every JavaScript module can import its own dependencies. When the browser
+receives a module, all `import` statements will trigger additional fetches to get ahold of the
+modules that are necessary to run the code. However, all these modules can have dependencies of
+their own. The danger is that the browser ends up with a cascade of fetches that lasts for multiple
+round-trip times (RTT) before the code can finally be executed.
+
+**ADD IMAGE for loose modules**
+
+#### Bundling
+
+Inlining all your modules into one single bundle (“bundling”) will make sure the browser has all the
+code it needs after 1 RTT and can start running the code more quickly. This, however, forces the
+user to download a lot of code that is not needed, so bandwidth and time has been wasted.
+Additionally, every change to one of our original modules will result in a change in the bundle,
+invalidating any cached version of the bundle. Users will have to re-download the entire thing.
+
+#### Code splitting
+
+Code splitting is the middle ground. We are willing to invest additional RTTs to get network
+efficiency by only downloading what we need and better caching efficiency by making the number of
+modules per bundle much smaller. If the bundling is done right, the total number of RTTs will be
+much lower than with loose modules. Finally, we could make use of [pushing
+mechanisms](/web/fundamentals/performance/prpl-pattern/) like `link[rel=preload]` or HTTP/2 Push to
+save additional RTTs if needed.
+
+**ADD IMAGE for code splitting**
+
+### Step 1: Obtain a list of your entry points
+
+> Time stamp: 16:02
+
+This is not to be taken seriously, but in the episode we parsed the website’s
+[`sitemap.xml`](https://github.com/GoogleChrome/ui-element-samples/blob/gh-pages/code-splitting/app/sitemap.xml)
+to get the entry points to our website.
+
+### Using babel to process JavaScript
+
+> Time stamp: 25:25 to 28:25
+
+Babel is commonly used for “transpiling”: Consuming bleeding-edge JavaScript code and turning it
+into an older version of JavaScript so that more browsers are able to execute the code. The first
+step here is to parse the new JavaScript with a parser (Babel uses
+[babylon](https://github.com/babel/babylon)) that turns the code into a so-called “Abstrac Syntax
+Tree” (AST). Once the AST has been generated, a series of plugins analyze and mangle the AST.
+
+Note: I am not very experiences with Babel. The plugins I built work, but they are probably neither
+efficient not very babel idiomatic. For a more in-depth guide to authoring Babel plugins, I
```

Comment:
I think you mean 'nor' instead of 'not'. A better phrase would be 'nor idiomatically babel'.